models/title.go

```diff
@@ -80,12 +83,28 @@ func (t Title) HasFees() bool {
 	return t.Fees != nil
 }
 
-func (t *Title) HasDiscounts() bool {
+func (t Title) HasDiscounts() bool {
 	return t.Discounts != nil
 }
 
+func (t *Title) OrderDiscountsByDate() *Title {
```

Comment:
```suggestion
func (t *Title) SortDiscountsByDate() *Title {
```

Outra coisa é que não fica claro se a ordenação é crescente ou decrescente. Vale a pena deixar isso explícito.

Reply:
Hmm, pensei em colocar SortDiscountsByMostRecentDate. Pode ser?